src/components/Charts/Axis/Axis.js

```diff
@@ -834,15 +834,24 @@ export default class Axis extends Component {
   }
 
   shouldRenderAxisSelector = () => {
-    if (this.props.scale?.type === 'LINEAR' && this.props.scale?.allFields?.length <= 1) {
-      return false
-    } else if (this.props.scale?.type === 'BAND' && !this.props.isAggregated && this.props.scale?.allFields <= 1) {
-      return false
-    } else if (this.props.isAggregated && this.props.legendLocation) {
+    const { scale, isAggregated, legendLocation, originalColumns, columns } = this.props
+    const scaleType = scale?.type
+    const isStringAxis = scaleType === 'BAND' || scaleType === 'TIME'
+
+    // For numeric (LINEAR) axes
+    if (!isStringAxis) {
+      return scale?.allFields?.length > 1 || scale?.hasDropdown
+    }
+
+    // For string axes, show selector if aggregated with legend or multiple groupable columns
+    if (isAggregated && legendLocation) {
       return true
     }
 
-    return this.props.scale?.hasDropdown
```

Comment:
ideally there shouldn't be so much logic in this function. If we need to adjust when "hasDropdown" is true, then we should really be doing that in the utils function so we arent combining multiple functions with different logic in them